tests/io/test_readwrite.py

```diff
@@ -28,134 +27,198 @@
-
         for k, v in _get_images().items():
-            if isinstance(v, SpatialImage):
-                v.name = f"incremental_{k}"
-            elif isinstance(v, MultiscaleSpatialImage):
-                for scale in v:
-                    names = list(v[scale].keys())
-                    assert len(names) == 1
-                    name = names[0]
-                    v[scale] = v[scale].rename_vars({name: f"incremental_{k}"})
-            sdata.images[f"incremental_{k}"] = v
+            sdata.images[f"additional_{k}"] = v
             with pytest.warns(UserWarning):
-                sdata.images[f"incremental_{k}"] = v
-                sdata[f"incremental_{k}"] = v
+                sdata.images[f"additional_{k}"] = v
+                sdata[f"additional_{k}"] = v
+            with pytest.raises(KeyError, match="Key `table` already exists."):
+                sdata["table"] = v
 
         for k, v in _get_labels().items():
-            if isinstance(v, SpatialImage):
-                v.name = f"incremental_{k}"
-            elif isinstance(v, MultiscaleSpatialImage):
-                for scale in v:
-                    names = list(v[scale].keys())
-                    assert len(names) == 1
-                    name = names[0]
-                    v[scale] = v[scale].rename_vars({name: f"incremental_{k}"})
-            sdata.labels[f"incremental_{k}"] = v
+            sdata.labels[f"additional_{k}"] = v
             with pytest.warns(UserWarning):
-                sdata.labels[f"incremental_{k}"] = v
-                sdata[f"incremental_{k}"] = v
+                sdata.labels[f"additional_{k}"] = v
+                sdata[f"additional_{k}"] = v
+            with pytest.raises(KeyError, match="Key `table` already exists."):
+                sdata["table"] = v
 
         for k, v in _get_shapes().items():
-            sdata.shapes[f"incremental_{k}"] = v
+            sdata.shapes[f"additional_{k}"] = v
             with pytest.warns(UserWarning):
-                sdata.shapes[f"incremental_{k}"] = v
-                sdata[f"incremental_{k}"] = v
-            break
+                sdata.shapes[f"additional_{k}"] = v
+                sdata[f"additional_{k}"] = v
+            with pytest.raises(KeyError, match="Key `table` already exists."):
+                sdata["table"] = v
 
         for k, v in _get_points().items():
-            sdata.points[f"incremental_{k}"] = v
+            sdata.points[f"additional_{k}"] = v
             with pytest.warns(UserWarning):
-                sdata.points[f"incremental_{k}"] = v
-                sdata[f"incremental_{k}"] = v
-            break
+                sdata.points[f"additional_{k}"] = v
+                sdata[f"additional_{k}"] = v
+            with pytest.raises(KeyError, match="Key `table` already exists."):
+                sdata["table"] = v
+
+    @pytest.mark.parametrize("dask_backed", [True, False])
+    @pytest.mark.parametrize("workaround", [1, 2])
+    def test_incremental_io_on_disk(
```

Comment:
this should 💯 be a tutorial

Reply:
I agree; for the moment I refer to this code in this GitHub conversation: https://github.com/scverse/spatialdata/discussions/520. Also, in the models notebooks I refer to the same GitHub conversation. The user would have to jump a bit but at least this code is discoverable.